### PR TITLE
Docs to split sharded database

### PR DESCRIPTION
**Pull request by emord**
##### SUMMARY
Document splitting a shard.

##### ENVIRONMENTS AFFECTED
icds

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
postgres

##### ADDITIONAL INFORMATION
Rough draft of a start. Used https://github.com/dimagi/commcare-cloud/blob/a6a32db56fb88314e45b5bc8f4cd748e08241de4/docs/services/postgresql/move-partitioned-database.md as a starting point

### CODE REVIEW

File: docs/services/postgresql/split-partitioned-database.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+# Splitting a shard in postgresql
+
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]: Should add introduction

File: docs/services/postgresql/split-partitioned-database.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+### 2. Stop all DB requests
+This will bring the CommCare HQ site down.
+
+**Stop all CommCare processes**
+```bash
+commcare-cloud <env> downtime start
+```
+
+**Stop pgbouncer**
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]: I think there is probably a better commcare-cloud command for this

File: docs/services/postgresql/split-partitioned-database.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+commcare-cloud <env> django-manage print_approximate_doc_distribution --csv
+```
+
+### 4. Update database name
+
+In pg2 postgres shell:
+
+```sql
+ALTER DATABASE commcarehq_p1 RENAME TO commcarehq_p2;
+```
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]: There's currently no way to manually specify a database name for a sharded database. It must be `commcarehq_django_alias`.

File: docs/services/postgresql/split-partitioned-database.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+       ...
+```
+
+**Deploy changes**
+```
+# update localsettings
+commcare-cloud <env> update-config
+
+# update PostgreSQL config on new PG node
+commcare-cloud <env> ap deploy_db.yml --limit=pg2
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]: perhaps this should be both pg1 & pg2

File: docs/services/postgresql/split-partitioned-database.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+Output shows that master and standby are up to date (both processing the same log).
+
+**Promote *pg2***
+```bash
+commcare-cloud <env> run-shell-command pg2 -b 'pg_ctlcluster <pg version> main promote'
+```
+
+
+### 8. Verify doc counts
+Re-run command from step 5 to verify that the document counts are double the number before.
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]:
```suggestion
Re-run command from step 6 to verify that the document counts are double the number before.
```

File: docs/services/postgresql/split-partitioned-database.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+Optionally you can run a vacuum to more reclaim space from the table.
+
+```sql
+VACUUM FULL table_name;
+```
+
+### 11. Verify doc counts
+Re-run command from step 5 to verify that the document counts are the same number as before.
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]:
```suggestion
Re-run command from step 6 to verify that the document counts are the same number as in step 5 (half as in step 8).
```

File: docs/services/postgresql/split-partitioned-database.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+**Unpause pgbouncer**
+```bash
+pg1 $ psql -p 6543 -U someuser pgbouncer
+
+> RESUME commcarehq_p1
+```
+
+**Restart services**
+```bash
+commcare-cloud <env> fab supervisorctl:"start all"
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]:
```suggestion
commcare-cloud <env> downtime end
```

File: docs/services/postgresql/split-partitioned-database.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+### 13. Validate the setup
```
Comment (emord) [inline docs/services/postgresql/split-partitioned-database.md]: should likely delete this?

File: docs/services/postgresql/split-partitioned-database.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+
+**Stop all CommCare processes**
+```bash
+commcare-cloud <env> downtime start
+```
+
+**Stop pgbouncer**
+
+```bash
+commcare-cloud <env> run-shell-command 'monit stop pgbouncer' --become
```
Comment (snopoke) [inline docs/services/postgresql/split-partitioned-database.md]: or `commcare-cloud <env> service postgres stop --only pgbouncer`

File: docs/services/postgresql/split-partitioned-database.md (hunk 3 above)
Comment (snopoke) [reply docs/services/postgresql/split-partitioned-database.md]: sure but the name comes from the Django DB settings right? I mean I would still want to rename the DB.

File: docs/services/postgresql/split-partitioned-database.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+```
+commcare-cloud <env> django-manage print_approximate_doc_distribution --csv
+```
+
+### 4. Update database name
+
+In pg2 postgres shell:
+
+```sql
+ALTER DATABASE commcarehq_p1 RENAME TO commcarehq_p2;
```
[the remark this answers is not in the document]
Comment (emord) [reply docs/services/postgresql/split-partitioned-database.md]: ALso needs to update pgbouncer config after this

**Comment by emord** [comment]
closing because we're changing to pglogical

**Pull request closed without merging**